### PR TITLE
Refactor footer logos to include mandatory link [BE]

### PR DESCRIPTION
**Pull request by engineervix**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1399154333)

### Description of Changes Made

Based on the feedback received during the Sprint 3 demo today, this PR modifies the footer logos to include links, which are mandatory.

The link title is used as the logo alt text and anchor title attribute.

### How to Test

- In Wagtail Admin, go to [Settings > Navigation Settings](http://127.0.0.1:8000/admin/settings/navigation/navigationsettings/1/) and add a couple of footer logos.
- Attempting to save the settings without specifying a link should raise a validation error
- When you add the logos, they should be displayed on the template. When you put your mouse on the image, you should see the text for the link title (if a title wasn't specified, that is, for an internal link, then you should see the page title), and the image should be a clickable link.

### Screenshots

<details><summary>Wagtail Admin</summary>
  
image

</details> 

<details>
  <summary>Validation Error</summary>

image

</details>

<details><summary>Template</summary>

image

</details> 

<details><summary>Pattern Library</summary>

image

</details> 

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on Ubuntu
  - Latest version of Firefox on Ubuntu

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] Not required

#### Pattern library

- [x] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

### CODE REVIEW

**Review by ianmeigh — COMMENTED** [review]
Looks good to me, @engineervix.

File: tbx/project_styleguide/templates/patterns/organisms/footer/footer.html [hunk 1]
```diff
@@ -42,13 +42,17 @@
 
     <div class="grid footer__bottom">
         <div class="grid__footer-logos">
-            {% with global_settings=settings.navigation.NavigationSettings %}
+            {% with settings.navigation.NavigationSettings as global_settings %}
```
Comment (nimasmi) [inline tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:45]: **Question** (completely opinionated and optional): Why revert to the old format? The new one permits setting multiple variables, and to me seems more familiar and easier to read as it mimics the appearance of variable assignment in code.

> The previous more verbose format is still supported: `{% with business.employees.count as total %}`  
> —https://docs.djangoproject.com/en/5.0/ref/templates/builtins/#with

The new syntax was introduced in Django 1.3 in 2011, so I can understand that we've not had much time to adjust: https://docs.djangoproject.com/en/5.0/releases/1.3/#improvements-to-built-in-template-tags

**Review by nimasmi — APPROVED** [review]

File: tbx/project_styleguide/templates/patterns/organisms/footer/footer.html (hunk 1 above)
Comment (engineervix) [reply tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:45]: Thanks @nimasmi

I changed to this while trying to make it work in the pattern library. For some reason, the pattern library component only rendered correctly when I changed to this syntax. I'll check again and see if there's a way I can use the new syntax without breaking the pattern library.

Comment (helenb) [reply tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:45]: don't spend too long worrying about that detail @engineervix - better to get this merged and focus on new stuff.

Comment (nimasmi) [reply tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:45]: Oh, curious. Thanks for explaining. Might this be something we need to feed back to the pattern library project?

Comment (engineervix) [reply tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:45]: Once I check and confirm that I wasn't just using the wrong syntax in the YAML file,then I can raise it with the django-pattern-library project :+1:

**Pull request merged**